### PR TITLE
fix(city-guide): hide empty filter pills instead of disabling them

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [FIREWORKS-59] (sub-issue 2)

### Description

Product decision from PM Viviana Flores, via a Notion comment on the FIREWORKS-59 QA bug:

> if there's no content for pills (i.e. fairs or saved), then those filter pills should not be displayed on the map.

This resolves the "disabled pills give no clear affordance, too visually subtle" complaint by never rendering an empty-content pill at all, instead of rendering it disabled at the end of the row.

- `CityFilterPills.tsx`: replaced the "tabs with results lead, empty ones pushed to the end and disabled" logic with a filter that drops any tab with no content. The "All" pill always renders, since it's the selection anchor for the row.
- `MapView.tsx`'s separate itinerary-section-filter pills (`filterableSections`/`pills`) already filtered out sections with zero places — confirmed and left unchanged, no work needed there.
- Updated `CityFilterPills.tests.tsx` to cover a zero-result tab not rendering at all (including the all-tabs-empty edge case, where only "All" remains).

### PR Checklist

- [x] I have tested my changes on the following platforms:
  - [ ] **Android**.
  - [ ] **iOS**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos** at least on **Android**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Fix: city guide map filter pills with no content are now hidden instead of shown disabled

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

🤖 Generated with [Claude Code](https://claude.com/claude-code)

Assisted-by: Claude:Sonnet-5

[FIREWORKS-59]: https://artsyproduct.atlassian.net/browse/FIREWORKS-59?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by artsyit** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Fix: city guide map filter pills with no content are now hidden instead of shown disabled - MounirDhahri)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f586b009dec399c07c0f71ef72a84b0c0815a830
</p>

File: src/app/Scenes/CityGuide/Components/CityFilterPills.tsx [hunk 1]
```diff
@@ -43,21 +43,13 @@ export const CityFilterPills: React.FC<CityFilterPillsProps> = ({
...
-
-    return [...withResults, ...withoutResults]
-  }, [bucketResults])
+  // Tabs with no content aren't shown at all, since a disabled pill reads as barely-there
+  // rather than intentionally hidden. "All" always renders, since it's the selection anchor
+  // and its own content mirrors what's already on the map.
+  const visibleTabs = useMemo(
+    () => cityTabs.filter((tab) => tab.id === "all" || tabHasResults(tab, bucketResults)),
+    [bucketResults]
+  )
```
[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/app/Scenes/CityGuide/Components/CityFilterPills.tsx:52]: Good catch. Fixed in 84fae0d: `CityFilterPills` now computes an effective selected id that falls back to `"all"` when `selectedTabId` isn't in `visibleTabs` (e.g. the drawer/pager selected a now-empty tab), so the pill row always shows a sensible highlight instead of none. Added a test covering selecting an empty tab via the `onSelectTab` prop path.

**Comment by artsyit** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Fix: city guide map filter pills with no content are now hidden instead of shown disabled - MounirDhahri)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 84fae0dea7923d018ba0742e1c142cf2b68d6ffe
</p>
